src/main/java/com/google/step/YTLounge/servlets/KeywordSearch.java

```diff
@@ -0,0 +1,105 @@
+package com.google.step.YTLounge.servlets;
+
+import com.google.api.client.googleapis.javanet.GoogleNetHttpTransport;
+import com.google.api.client.http.javanet.NetHttpTransport;
+import com.google.api.client.json.JsonFactory;
+import com.google.api.client.json.jackson2.JacksonFactory;
+import com.google.api.services.youtube.YouTube;
+import com.google.api.services.youtube.YouTubeRequestInitializer;
+import com.google.api.services.youtube.model.SearchListResponse;
+import com.google.gson.Gson;
+import com.google.step.YTLounge.data.RequestParameter;
+import java.io.File;
+import java.io.IOException;
+import java.security.GeneralSecurityException;
+import java.util.Scanner;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+@WebServlet("/keywordSearch")
+public class KeywordSearch extends HttpServlet {
+  private static final JsonFactory jsonFactory = JacksonFactory.getDefaultInstance();
+
+  /**
+   * Build and return an authorized API client service.
+   *
+   * @return an authorized API client service
+   * @throws GeneralSecurityException, IOException
+   */
+  public YouTube getService() throws GeneralSecurityException, IOException {
+    String key = readSecrets();
+    final YouTubeRequestInitializer keyInitializer = new YouTubeRequestInitializer(key);
+    final NetHttpTransport httpTransport = GoogleNetHttpTransport.newTrustedTransport();
+    return new YouTube.Builder(httpTransport, jsonFactory, null)
+        .setApplicationName("YouTube Lounge")
+        .setYouTubeRequestInitializer(keyInitializer)
+        .build();
+  }
+
+  @Override
+  public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    Gson gson = new Gson();
+    response.setContentType("application/json");
+    String keywords = RequestParameter.getParameter(request, "words", "");
+    YouTube youtubeService = null;
+    try {
+      youtubeService = getService();
+    } catch (Exception e) {
+      e.printStackTrace();
+    }
+    if (youtubeService == null) {
+      return;
+    }
+    // Define and execute the API request
+    YouTube.Search.List vidRequest = youtubeService.search().list("snippet");
+    SearchListResponse vidResponse =
+        vidRequest.setMaxResults(15L).setQ(keywords).setType("video").execute();
+    response.getWriter().println(gson.toJson(vidResponse));
+  }
+
+  /**
+   * Takes the given string and converts it into pure seconds. String must be either in the format
+   * of "PT#H#M#S" or "PT#M#S" to be properly parsed.
+   *
+   * @return a long for the number of seconds in encodedTime
+   */
+  public long parseDuration(String encodedTime) {
```

Comment:
I remember reviewing similar function before. Was that for JavaScript?

Reply:
I thought I would need this function for this servlet, but it isn't necessary